source/core/Array.cpp

```diff
@@ -127,6 +128,66 @@ VIREO_FUNCTION_SIGNATUREV(ArrayIndexEltNDV, ArrayIndexNDVParamBlock)
     }
     return _NextInstruction();
 }
+
+//indexing on 2d array, the row argument comes before column argument
+//which is different from the indexing used in normal N dimension functions(e.g 0 d index, 1st d index, 3nd d index ...)
+VIREO_FUNCTION_SIGNATURE6(ArrayIndexElt2DV, TypedArrayCoreRef, StaticType, void, StaticType, void, void)
+{
+    TypedArrayCoreRef arrayIn = _Param(0);
+    Boolean pickRow = false;
+    Boolean pickCol = false;
+    IntIndex row, col;
+    IntIndex rank = 2;
+    TypeRef type = _ParamPointer(1);
+    IntIndex* lengths = arrayIn->DimensionLengths();
+    if (type->IsA("Int32")) {
```

Comment:
Should no need to check type at runtime for indexes. If just checking for "unwired" optional parameter then simply check data pointer for null.